### PR TITLE
feat: upgrade to node v20

### PR DESCRIPTION
**Pull request by saikumarrs**
## What are the changes introduced in this PR?

Upgrade Node to v20.
Also, I fixed the error message format in the DNS lookup module.

Additionally, I added a Node option (`--no-node-snapshot`) to avoid segmentation faults while running the service in specific machines ([ref](https://github.com/laverdet/isolated-vm/issues/424#issuecomment-1864629126)).

## What is the related Linear task?

Resolves INT-1547

## Please explain the objectives of your changes below

To the latest Node LTS version.

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

Since the Node environment has been upgraded, we need to do some performance testing.

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [-] All related docs linked with the PR?

- [-] All changes manually tested?

- [-] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3069/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3069/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3069/test-report.html

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # syntax=docker/dockerfile:1.4
-FROM node:18.19.0-alpine3.18 AS base
+FROM node:20.11.0-alpine3.19 AS base
```
Comment (koladilip) [inline Dockerfile:2]: should we use 20-alpine instead?

**Review by koladilip — DISMISSED** [review]

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3069/test-report.html

File: Dockerfile (hunk 1 above)
Comment (saikumarrs) [reply Dockerfile:2]: I would recommend specific versions as we can be less prone to future vulnerabilities and have more control.

We can frequently upgrade this dependency along with the others like NPM packages.

Comment (koladilip) [reply Dockerfile:2]: we need to plan node upgrades more often than

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3069/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3069/test-report.html

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Comment by saikumarrs** [comment]
Deprioritized. Hence, closing it.

**Pull request closed without merging**